miraclelinux/meta-debian
# openssh: Update version from 7.9p1-10+deb10u3 to 7.9p1-10+deb10u4

# Purpose of pull request
This version fixes following CVEs.
- CVE-2021-41617
- CVE-2023-48795
- CVE-2023-51385

DLA: DLA 3694-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. (optional) Prepare for build
2. Build qemuarm64 image
3. Run qemuarm64 image
4. Prepare for run ptest
5. Run ptest and check ptest log

### (optional) Prepare for build
Set MIRROR to local.conf if necessary.
```
MIRRORS += "${DEBIAN_SECURITY_UPDATE_MIRROR} https://snapshot.debian.org/archive/debian-security/20231218T230659Z/pool/updates/ \n"
```

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
DISTRO_FEATURES_append = " pam"
IMAGE_INSTALL_append = " openssh-keygen openssh-misc openssh-scp openssh-sftp openssh-sftp-server openssh-ssh openssh-sshd"
IMAGE_INSTALL_append = " openssh-ptest"
```
```
$ bitbake core-image-minimal
```

### Run qemuarm64 image
```
$ runqemu qemuarm64 nographic
```

### Prepare for run ptest
The openssh's ptest fails because the localhost name cannot be resolved. (See [issue#332](https://github.com/meta-debian/meta-debian/issues/332))
Add /etc/hosts file for localhost.
```
# echo "127.0.0.1 localhost" >> /etc/hosts
```

### Run ptest and check ptest log
The command to display ptest results was created with reference to meta-debian's tests/run_ptest.sh.
```
# test_package=openssh
# ptest-runner $test_package 2>&1 | tee ptest.log
# echo "$test_package: PASS/SKIP/FAIL = `grep '^PASS:' ptest.log | wc -l`/`grep '^SKIP:' ptest.log | wc -l`/`grep '^FAIL:' ptest.log | wc -l`"
```

# Test result
## Build package
Build succeeded.
```
$ bitbake openssh
Parsing recipes: 100% |#######################################################################################| Time: 0:00:40
Parsing of 1041 .bb files complete (0 cached, 1041 parsed). 1821 targets, 73 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "debian-10"
TARGET_SYS           = "aarch64-deby-linux"
MACHINE              = "qemuarm64"
DISTRO               = "deby"
DISTRO_VERSION       = "10.0"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-poky            = "warrior:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-openssh_7.9p1-10+deb10u4:d7b814305dff8bb31fbfb17588828d9b8bf10a89"

Initialising tasks: 100% |####################################################################################| Time: 0:00:07
Sstate summary: Wanted 428 Found 0 Missed 428 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 1995 tasks of which 0 didn't need to be rerun and all succeeded.
```

## Package test
The ptest has passed (no FAILs).
```
# echo "127.0.0.1 localhost" >> /etc/hosts
# ptest-runner $test_package 2>&1 | tee ptest.log
START: ptest-runner
2024-01-09T09:07
BEGIN: /usr/lib/openssh/ptest
...
PASS: authinfo
SKIP: agent-ptrace
DURATION: 4310
END: /usr/lib/openssh/ptest
2024-01-09T10:19
STOP: ptest-runner
# echo "$test_package: PASS/SKIP/FAIL = `grep '^PASS:' ptest.log | wc -l`/`grep '^SKIP:' ptest.log | wc -l`/`grep '^FAIL:' ptest.log | wc -l`"
openssh: PASS/SKIP/FAIL = 66/7/0
```
